docs/app/____-represent-flows-as-pushdown-automata.md

```diff
@@ -0,0 +1,71 @@
+# Defining an Abstraction for Application Flows as Pushdown Automata
+
+## Context
+
+We've previously been using finite state machines to represent application flows. This has provided a number of benefits:
+
+1. Provides a declarative way to define page transitions, which reduces bugs by constraining what is possible to a less expressive language
+2. Provides a core data structure that can be augmented with additional functionality, such as visualization
+
+Implementation-wise, we've been using the [xstate library](https://xstate.js.org/).
+
+## Problem
```

Comment:
Thanks for writing this up! These charts make me interested to learn more about pushdown automaton for sure. I don't have a solid grasp of what the solution to these problems looks like in code, from reading this doc, but I don't know if that was necessarily the goal. 

I'm not fully following yet how the state, like the application state, could be used for the conditional routing logic.

It seems like the `emit` part of this implementation is where logic [the "guards" in PFML](https://github.com/navapbc/archive-massgov-pfml/blob/18036249b50fedd981f31c0d796c026bbdefb8a2/portal/src/flows/claimant.ts#L124) would live, and after running the guards, it would return a single(?) "event" which is then used for determining which of the next routes is used?

Reply:
The "emit" part was me handwaving the visualization. I think it might be clearer if I just got rid of the "emit" word and just labeled it as "authenticated". The point I was making was that the same event "authenticated", can change the state in two different state machines i.e. the one up top and the one in the AdditionalState section too. Maybe I could take a pass at adding some implementation pseudocode to make it more concrete.

Reply:
I find pseudocode really helpful 👍🏻

Reply:
Took a pass at defining the config as code. Take a look and lemme know what you think.

Reply:
The general idea is that we can define rules in some sort of order, so the machine basically goes through the rules to figure out which rule to apply. Something like that. We might want to separate out the "additional state rule" from the "flow rules" so that they can be applied separately. Idea is still pretty rough.